src/write.jl

```diff
@@ -39,33 +49,29 @@ function set_value(ptrentry, tagv)
     elseif entry.format == LibExif.EXIF_FORMAT_LONG
         LibExif.exif_set_long(entry.data, FILE_BYTE_ORDER, parse(Cuint, tagv))
     elseif entry.format == LibExif.EXIF_FORMAT_RATIONAL
-        if entry.tag in [LibExif.EXIF_TAG_FNUMBER, LibExif.EXIF_TAG_APERTURE_VALUE, LibExif.EXIF_TAG_MAX_APERTURE_VALUE]    
+        if entry.tag in (LibExif.EXIF_TAG_FNUMBER, LibExif.EXIF_TAG_APERTURE_VALUE, LibExif.EXIF_TAG_MAX_APERTURE_VALUE)    
             p = Rational(parse(Float32, split(tagv, "/")[2]))
         else
             p = rationalize(parse(Float32, tagv);tol=0.1)
         end
         LibExif.exif_set_rational(entry.data,FILE_BYTE_ORDER, LibExif.ExifRational(p.num, p.den))
     elseif entry.format == LibExif.EXIF_FORMAT_ASCII
-        len = sizeof(tagv)+1
-        unsafe_store!(ptrentry.size, Cuint(len), 1)
-        unsafe_store!(ptrentry.components, Culong(len), 1)
-        mem = LibExif.exif_mem_new_default()
-        buf = LibExif.exif_mem_alloc(mem, len)
-        unsafe_copyto!(buf, pointer(Vector{UInt8}(tagv * "\0")), len)
-        ptrentry.data = buf
+        ptrentry.data = get_ascii_buffer(ptrentry, tagv)
     elseif entry.format == LibExif.EXIF_FORMAT_SRATIONAL
         p = Rational(parse(Float32, tagv))
         LibExif.exif_set_srational(entry.data, FILE_BYTE_ORDER, LibExif.ExifSRational(p.num, p.den))
     elseif entry.format == LibExif.EXIF_FORMAT_UNDEFINED
         if entry.tag == LibExif.EXIF_TAG_FLASH_PIX_VERSION
             data = Dict{String,String}(
-            "FlashPix Version 1.0" => "0100\0",
-            "FlashPix Version 1.01" => "0101\0",
-            "Unknown FlashPix Version" => "0000\0",
+                "FlashPix Version 1.0" => "0100\0",
+                "FlashPix Version 1.01" => "0101\0",
+                "Unknown FlashPix Version" => "0000\0",
             )
             unsafe_copyto!(entry.data, pointer(Vector{UInt8}(data[tagv])), 5)
+        elseif entry.tag == LibExif.EXIF_TAG_USER_COMMENT
+            ptrentry.data = get_ascii_buffer(ptrentry, tagv)
         else
-            @debug "Tag unsupported" entry.tag
+            @debug "Tag unsupported (EXIF_FORMAT_UNDEFINED)" entry.tag
```

Comment:
Better debug messages! albeit I wonder should I make these `@info` to inform of unsupported tags. This caused an issue for you I think where you couldn't identify where that comment tag was disappearing.

Reply:
I should have read the source code first and set `JULIA_DEBUG=ExifViewer` when hitting  https://github.com/JuliaImages/ExifViewer.jl/issues/24 ...
It's always a tradeoff for end users not wanting spurious warnings.
But silently ignoring is confusing too, so I don't have a strong opinion on that one (appart from disambiguating the `@debug "Tag unsupported"` message from the one a few lines below).